arroyo-sql/src/types.rs

```diff
@@ -901,6 +901,16 @@ impl TypeDef {
         }
     }
 
+    pub fn can_be_key(&self) -> bool {
```

Comment:
Can we have this be a `Result<()>` which reports that float can't be a key, rather than having the callers report that?

Reply:
That's a good idea